pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller.go

```diff
@@ -29,6 +34,80 @@ import (
 	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/nsxserviceaccount"
 )
 
+var (
+	proxyServicePred = predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*corev1.Service)
+			newObj := e.ObjectNew.(*corev1.Service)
+
+			newLabels := e.ObjectNew.GetLabels()
+			oldLabels := e.ObjectOld.GetLabels()
+			var oldObjHasLabel, newObjHasLabel bool
+			if val, ok := oldLabels[proxyLabelKey]; ok && val == "" {
```

Comment:
when will val not equal to ""?

Reply:
Currently, it will always be "". But it can be set to some value manually be user via kubectl on supervisor.

Reply:
I think we do not care about what the val is, just think that we may remove the val directly